src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -406,24 +407,42 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
          1. RequestPriority
          2. Global workqueue statistics, while acquiring a workflow
         """
-        if 'RequestPriority' in request_args:
-            # Yes, we completely ignore any other arguments posted by the user (web UI case)
-            request_args = {'RequestPriority': request_args['RequestPriority']}
-            validate_request_priority(request_args)
-            # must update three places: GQ elements, workload_cache and workload spec
-            self.gq_service.updatePriority(workload.name(), request_args['RequestPriority'])
-            report = self.reqmgr_db_service.updateRequestProperty(workload.name(), request_args, dn)
-            workload.setPriority(request_args['RequestPriority'])
-            workload.saveCouchUrl(workload.specUrl())
-            cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), request_args['RequestPriority']))
-        elif workqueue_stat_validation(request_args):
-            report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), request_args))
-        else:
-            msg = "There are invalid arguments for no-status update: %s" % request_args
+        report = []
+        reqArgsDiff = deepcopy(request_args)
+
+        if not reqArgsDiff:
+            cherrypy.log("Nothing to be changed at this stage")
+            return reduceReport(report)
+
+        reqArgsNothandled = []
+        for reqArg in reqArgsDiff.copy():
```

Comment:
Iterating over a dictionary with the `copy()` method looks awkward. It looks like the dictionary is not changing size, so why not stick with the common pattern of `for reqArg in reqArgsDiff:` or `for reqArg in reqArgsDiff.keys():`?
If it is changed, then I would suggest `for reqArg in list(reqArgsDiff):`, such that it is more consistent with the rest of the codebase.

Reply:
a dictionary size  changes - we remove the `RequestName`  inside this cycle.

Reply:
I failed to see where `RequestName` is removed inside this loop. Can you please point me to the line/function doing so?

Reply:
Sorry, my bad. I already lost track of the code state when this copy of dictionary was needed, maybe the dictionary size was not changing due to the removal of `RequestName` but I suppose it may have been  changing because of the call to `workqueue_stat_validation`, which previously was inside the loop. Anyway, as I already said it matters not any more.